engine/runtime/src/main/scala/org/enso/interpreter/runtime/IrToTruffle.scala

```diff
@@ -2171,9 +2181,17 @@ private[runtime] class IrToTruffle(
 
       def args(): Array[ArgumentDefinition] = slots._2
       def bodyNode(): RuntimeExpression = {
-        val body = BlockNode.buildRoot(Array(), argsExpr._2)
+        var operation = argsExpr._2
+        if (!operation.isInstanceOf[BlockNode]) {
+          operation = BlockNode.buildStatements(Array(), operation)
+        }
+        val body = if (defineRoot) {
+          BlockNode.buildRoot(Array(), operation)
+        } else {
+          BlockNode.buildInvisible(Array(), operation)
+        }
         val initVariablesAndThenBody =
-          BlockNode.buildSilent(argsExpr._1.toArray, body)
+          BlockNode.buildInvisible(argsExpr._1.toArray, body)
```

Comment:
```suggestion
          BlockNode.buildInvisible(argsExpr._1, body)
```
redundant - `argsExpr._1` is already `Array`